robocorp-code/vscode-client/src/pythonExtIntegration.ts

```diff
@@ -43,26 +60,77 @@ export async function autoUpdateInterpreter(docUri: Uri) {
             ConfigurationTarget.WorkspaceFolder
         );
     }
+    return true;
 }
 
-export async function installPythonInterpreterCheck(context: ExtensionContext) {
-    context.subscriptions.push(
-        window.onDidChangeActiveTextEditor(async (event: TextEditor) => {
-            try {
-                // Whenever the active editor changes we update the Python interpreter used (if needed).
-                let docUri = event?.document?.uri;
-                if (docUri) {
-                    await autoUpdateInterpreter(docUri);
+export async function installWorkspaceWatcher(context: ExtensionContext) {
+    // listen to editor change/switch (this should cause environment switching as well)
+    const checkEditorSwitch = window.onDidChangeActiveTextEditor(async (event: TextEditor) => {
+        try {
+            // Whenever the active editor changes we update the Python interpreter used (if needed).
+            let docURI = event?.document?.uri;
+            if (docURI) {
+                await autoUpdateInterpreter(docURI);
+            }
+        } catch (error) {
+            logError("Error auto-updating Python interpreter.", error, "PYTHON_SET_INTERPRETER");
+        }
+    });
+
+    // listen for document changes and mark targeted files as dirty
+    const checkIfFilesHaveChanged = workspace.onDidChangeTextDocument(async (event) => {
+        let docURI = event.document.uri;
+        if (event.document.isDirty && (isEnvironmentFile(docURI.fsPath) || isPythonFile(docURI.fsPath))) {
+            dirtyWorkspaceFiles.add(docURI.fsPath);
+        }
+    });
+
+    // listen for when documents are about to be saved
```

Comment:
This comment: `// listen for when documents are about to be saved` is no longer true here.